backend/danswer/search/pipeline.py

```diff
@@ -114,47 +276,70 @@ def retrieved_docs(self) -> list[InferenceChunk]:
             retrieval_metrics_callback=self.retrieval_metrics_callback,
         )
 
-        # self._retrieved_docs = chunks_to_search_docs(retrieved_chunks)
-        return cast(list[InferenceChunk], self._retrieved_docs)
+        return cast(list[InferenceChunk], self._retrieved_chunks)
+
+    @property
+    def retrieved_sections(self) -> list[InferenceSection]:
+        # Calls retrieved_chunks inside
+        self._retrieved_sections = self._combine_chunks(post_rerank=False)
+        return self._retrieved_sections
 
     """Post-Processing"""
 
     @property
-    def reranked_docs(self) -> list[InferenceChunk]:
-        if self._reranked_docs is not None:
-            return self._reranked_docs
+    def reranked_chunks(self) -> list[InferenceChunk]:
+        if self._reranked_chunks is not None:
+            return self._reranked_chunks
 
         self._postprocessing_generator = search_postprocessing(
             search_query=self.search_query,
-            retrieved_chunks=self.retrieved_docs,
+            retrieved_chunks=self.retrieved_chunks,
             rerank_metrics_callback=self.rerank_metrics_callback,
         )
-        self._reranked_docs = cast(
+        self._reranked_chunks = cast(
             list[InferenceChunk], next(self._postprocessing_generator)
         )
-        return self._reranked_docs
+        return self._reranked_chunks
 
     @property
-    def relevant_chunk_indicies(self) -> list[int]:
-        if self._relevant_chunk_indicies is not None:
-            return self._relevant_chunk_indicies
+    def reranked_sections(self) -> list[InferenceSection]:
+        # Calls reranked_chunks inside
+        self._reranked_sections = self._combine_chunks(post_rerank=True)
+        return self._reranked_sections
+
+    @property
+    def relevant_chunk_indices(self) -> list[int]:
+        # If chunks have been merged, then we cannot simply rely on the leading chunk
+        # relevance, there is no way to get the full relevance of the Section now
+        # without running a more token heavy pass. This can be an option but not
+        # implementing now.
+        if self.ran_merge_chunk:
+            return []
+
+        if self._relevant_chunk_indices is not None:
+            return self._relevant_chunk_indices
 
         # run first step of postprocessing generator if not already done
-        reranked_docs = self.reranked_docs
+        reranked_docs = self.reranked_chunks
 
         relevant_chunk_ids = next(
             cast(Generator[list[str], None, None], self._postprocessing_generator)
         )
-        self._relevant_chunk_indicies = [
+        self._relevant_chunk_indices = [
             ind
             for ind, chunk in enumerate(reranked_docs)
             if chunk.unique_id in relevant_chunk_ids
         ]
-        return self._relevant_chunk_indicies
+        return self._relevant_chunk_indices
 
     @property
     def chunk_relevance_list(self) -> list[bool]:
```

Comment:
maybe split this into two functions, `chunk_relevance_list` and `section_relevance_list`